src/shogun/base/AnyParameter.h

```diff
@@ -28,101 +28,85 @@ namespace shogun
 		GRADIENT_AVAILABLE = 1
 	};
 	/** @brief Class AnyParameterProperties keeps track of parameter properties.
-	 *
-	 * Currently there are three types of parameters:
-	 * -# HYPERPARAMETER: the parameter determines how the training is
-	 * performed, e.g. regularisation
-	 * -# GRADIENT_PARAM: the parameter is used in gradient updates
-	 * -# MODEL_PARAM: model parameter that is trained, e.g. weight and bias
-	 * vectors and bias
+	 * The parameter properties can be either true or false.
```

Comment:
like!
Maybe also mention descriptions in the first sentence " ... of parameter meta information, such as properties and descriptions"